datajoint/s3.py

```diff
@@ -4,72 +4,62 @@
+
+    def get(self, relative_name):
+        return self.client.get_object(self.bucket, '/'.join((self.remote_path, relative_name))).data
+
+    def fget(self, relative_name, local_filepath):
+        """get file from object name to local filepath"""
+        name = '/'.join((self.remote_path, relative_name))
+        stat = self.client.stat_object(self.bucket, name)
+        meta = {k.lower().lstrip('x-amz-meta'): v for k, v in stat.metadata.items()}
+        data = self.client.get_object(self.bucket, name)
+        os.makedirs(os.path.split(local_filepath)[0], exist_ok=True)
+        with open(local_filepath, 'wb') as f:
+            for d in data.stream(1 << 16):
+                f.write(d)
+        return uuid.UUID(meta['contents_hash'])
+
+    def partial_get(self, relative_name, offset, size):
         try:
-            return self.client.get_partial_object(self.bucket, '/'.join((self.remote_path, blob_hash)), offset, size).data
+            return self.client.get_partial_object(
+                self.bucket, '/'.join((self.remote_path, relative_name)), offset, size).data
         except minio.error.NoSuchKey:
             return None
 
-    def get_size(self, blob_hash):
+    def get_size(self, relative_name):
         try:
-            return self.client.stat_object(self.bucket, '/'.join((self.remote_path, blob_hash))).size
+            return self.client.stat_object(self.bucket, '/'.join((self.remote_path, relative_name))).size
         except minio.error.NoSuchKey:
             return None
 
-    def clean(self, exclude, max_count=None, verbose=False):
-        """
-        Delete all objects except for those in the exclude
-        :param exclude: a list of blob_hashes to skip.
-        :param max_count: maximum number of object to delete
-        :param verbose: If True, print deleted objects
-        :return: list of objects that failed to delete
-        """
-        count = itertools.count()
-        if verbose:
-            def out(name):
-                next(count)
-                print(name)
-                return name
-        else:
-            def out(name):
-                next(count)
-                return name
-
-        if verbose:
-            print('Deleting...')
-
-        names = (out(x.object_name)
-                 for x in self.client.list_objects(self.bucket, self.remote_path + '/', recursive=True)
-                 if x.object_name.split('/')[-1] not in exclude)
+    def list_objects(self, folder=''):
+        return self.client.list_objects(self.bucket, '/'.join((self.remote_path, folder, '')), recursive=True)
 
-        failed_deletes = list(
-            self.client.remove_objects(self.bucket, itertools.islice(names, max_count)))
+    def remove_objects(self, objects_iter):
 
-        print('Deleted: %i S3 objects' % next(count))
-        return failed_deletes
+        failed_deletes = self.client.remove_objects(self.bucket, objects_iter=objects_iter)
+        return list(failed_deletes)
```

Comment:
Seems that there is a possibility here that a delete will trigger a delete of other existing external blobs that do not have `filepath` defined.

Reply:
Per discussion, this is correct but suggestion is that documentation should state that `filepath` data and `external blob` data are required to be in separate `stores`.